price_pusher/src/interface.ts

```diff
@@ -8,6 +8,7 @@ export type PriceInfo = {
 };
 
 export interface IPriceListener {
+  start(): Promise<void>;
```

Comment:
can you document the assumptions around this method? I think `start` must fetch the current prices before the promise returns, right?

Reply:
Also what happens if someone starts the class twice?

Reply:
it will setup another interval to poll prices
and another web socket connection 

Should we disallow it? I can add a static variable to check and return if it was already started

Reply:
we will discuss it later. this can be done in different pr